skyportal/handlers/api/taxonomy.py

```diff
@@ -161,10 +174,28 @@ def post(self):
         provenance = data.get('provenance', None)
 
         # compute the allowable classes
-        jsonpath_expr = parse('$..class')
-        allowed_classes = list(set(
-                               [x.value for x in jsonpath_expr.find(hierarchy)]
-                               ))
+        result = []
+
+        def dict_path(path, my_dict):
```

Comment:
Could you explain what these two paragraphs do?

Reply:
This generates the list of allowed classes for this hierarchy, capturing both the leaf nodes as well as the parent nodes as possible classes. In addition, it generates a label to be show during the selection process. Eg, the output is:
```
[{"class": "SN", "label": "<b>SN</br>⇦Cataclysmic"}, {"class": "Ia", "label": "<b>Ia</br>⇦SN⇦Cataclysmic"}]
```
The Classification selection on the Source page then uses array to allow the user to find the classification more easily using <Autocomplete> from MUI.

Reply:
Adding this response as a comment inline.